[Dataviews]: add storybook instance for this package

retrofox opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

This PR configures storybook for the dataviews package using the calypso-storybook default configuration. Updates package.json to include required dependencies and adds new script command.

**Removes dataviews exclusion from root package.json to allow proper workspace resolution.**
This change ensures the package is adequately recognized within the monorepo structure and can be built alongside other packages.

The setup includes basic configuration files (.storybook directory) with main.js and preview.js following our standard patterns for component documentation.

Related to #

Closes DOTCOM-6423

## Proposed Changes

* Add storybook instance for dataviews package

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

*

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

1) Update package dependencies
```sh
yarn
```
2) Run the Storybook app for this package
```sh
yarn dataviews:storybook:start
```
3) Confirm it runs the Storybook app

<img width="962" alt="image" src="https://github.com/user-attachments/assets/94c3b5d8-28e7-4535-8e86-9edf42cfd71d" />

4) Confirm you see the three stories in the client/browser

<img width="911" alt="image" src="https://github.com/user-attachments/assets/7cc4f898-1b44-4920-b0f3-9bc57049dab5" />

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

retrofox committed:
Add Storybook instance to dataviews package
Add Storybook configuration and dependencies with unique port 57863 for local development

retrofox committed:
register @automattic/dataviews package

retrofox committed:
add global script to run dataviews storybook app

retrofox committed:
rename story filename

retrofox committed:
set base styles support

retrofox committed:
import base styles in the Storybook context

retrofox committed:
Add sass math module to grid layout styles

retrofox committed:
Move react-dom to peer dependencies

oandregal commented on packages/dataviews/src/dataviews-layouts/grid/style.scss:
```diff
@@ -1,3 +1,5 @@
+@use "sass:math";
```
What does this change do? Why is it necessary?

retrofox replied:
Sorry forgot to mention in the PR description. It's required in a [few lines below](https://github.com/Automattic/wp-calypso/blob/5b1e6e4f92cf6e32f7fe96c2d152eb2a119f9336/packages/dataviews/src/dataviews-layouts/grid/style.scss#L87) when it calculates the padding-top.
I don’t know why it isn't imported in core. Especially because it is imported in other parts, [like here](https://github.com/WordPress/gutenberg/blob/a59d45e06d909138444815ab98f1c323066c9707/packages/components/src/form-toggle/style.scss#L1).

oandregal replied:
Removed this in favor of declaring it in storybook styles, see https://github.com/Automattic/wp-calypso/pull/102592#issuecomment-2801972957

oandregal commented on package.json:
```diff
@@ -10,7 +10,6 @@
 			"desktop",
 			"apps/*",
 			"packages/*",
-			"!packages/dataviews",
```
We can't do this yet, it's not ready.

retrofox replied:
It isn't ready for what? What are the risks of not registering the dataviews in the workspaces list?

retrofox replied:
Let me remove this change

retrofox replied:
reverted here https://github.com/Automattic/wp-calypso/pull/102592/commits/21ca8c206c0d7c56f92a710d24880a948c00a3cc

oandregal commented on packages/dataviews/package.json:
```diff
@@ -38,10 +38,14 @@
 	"wpScript": true,
 	"types": "build-types",
-	"sideEffects": false,
+	"sideEffects": [
```
What is this for?

retrofox replied:
We're adding sideEffects to the package.json to ensure CSS/SCSS files are properly loaded in Storybook while allowing webpack to optimize other modules through tree-shaking

oandregal replied:
Upstream does this differently, by loading the [built CSS files](https://github.com/WordPress/gutenberg/pull/56761/files). Can we change the storybook configuration to do the same and not changing `sideEffects: false` in this `package.json` file?

retrofox replied:
gotcha. Yes, let’s try to stick to the core den-env setup as much as possible

oandregal commented:
@retrofox is it necessary to rename the stories file? We aim to bring changes from upstream into `packages/dataviews`, so it's best that we do the minimal changes possible for subsequent syncs to be easier.

oandregal reviewed (changes requested):
I've left some comments for clarification, but the TLDR is that we can't land this yet if we're modifying code from `packages/dataviews`. Are there ways to land this without modifications to that code?

I'm going to mark this review as "request changes" so we don't merge it unintentionally.

retrofox commented:
<!-- linear:replyTo:2801333137 -->

😬 let me see if we can set up the app in Calypso

retrofox committed:
revert not excluding dataviews from workspace

retrofox committed:
DataViews: Add local yarn.lock for Storybook development
Enables running Storybook directly in the package without monorepo workspace configuration.

retrofox committed:
update Storybook start command to use local package
Changes the npm script to use --cwd instead of workspace for isolated package development.

retrofox committed:
DataViews: Add support for .story.tsx files in Storybook config
Extends stories pattern to include both .stories.tsx and .story.tsx file extensions.

retrofox commented:
> @retrofox is it necessary to rename the stories file? We aim to bring changes from upstream into `packages/dataviews`, so it's best that we do the minimal changes possible for subsequent syncs to be easier.

I've extended the config to support the core name convention for the stories here https://github.com/Automattic/wp-calypso/pull/102592/commits/f2e6a6b30005e22d0b2aac1925514e88b25d4ce1

oandregal commented:
@retrofox I've pushed a change at https://github.com/Automattic/wp-calypso/pull/102592/commits/642f36a96eb1291c678106452bd35a5c85254cf3 so that the sass/math module is loaded only for the storybook styles. Upstream, this is done in the process that compiles SASS into CSS ([see](https://github.com/WordPress/gutenberg/blob/b23709686bea6c64b33fd476437b0d5cfd68a382/bin/packages/build-worker.js#L126)). I'd like to address this separately.

retrofox committed:
tidy/janitorial

retrofox committed:
reintroduce sass:math dep

retrofox committed:
remove @babel/runtiem dep

retrofox committed:
remove dependency

retrofox committed:
import compiled styles

retrofox committed:
Fix styles loading in Storybook
Modified Storybook webpack configuration to ensure styles are properly loaded when `sideEffects` is disabled in the package. This change is needed because Storybook requires CSS/SCSS files to be treated as side effects to work correctly.

retrofox committed:
discard change removing babel/runtime

retrofox committed:
remove babel/runtime from dataviews package

retrofox committed:
remove package yarn files

retrofox committed:
update storybook deps

retrofox committed:
Move sass math module to storybook styles

retrofox committed:
fix srotybook root script

retrofox commented:
> I've left some comments for clarification, but the TLDR is that we can't land this yet if we're modifying code from `packages/dataviews`. Are there ways to land this without modifications to that code?
> 
> I'm going to mark this review as "request changes" so we don't merge it unintentionally.

PR updated and RfR again. :thanks!:

retrofox committed:
dedupe

retrofox committed:
use react 18.0.0 version

retrofox committed:
update yarn
